pele_platform/Utilities/Helpers/smiles_constraints.py

```diff
@@ -1,63 +1,68 @@
 from dataclasses import dataclass
 import re
 import pele_platform.Errors.custom_errors as ce
+import pele_platform.Utilities.Helpers.helpers as hp
+
 
 @dataclass
 class SmilesConstraints:
-
     input_pdb: str
     constrain_core: str
     resname: str
     chain: str
     spring_constant: float = 50.0
+    hp.is_rdkit()
 
     def run(self):
-        from rdkit import Chem
-        self._convert_to_smarts()
-        self._extract_ligand()
-        self._get_matches()
+        self.smarts = self._convert_to_smarts()
```

Comment:
self.smarts = self.convert_to_smarts(self.smiles) --> If we make the function to have an input we can make it public and let people use it!